File: Java/FunçõesJava/BoasVindas.java
import java.util.Scanner;

public class BoasVindas {

    // Versão 2 - Função (Modular)
    public static String gerarMensagemBoasVindas(String nome, int idade) {
        return "Seja bem-vindo(a), " + nome + "! Você tem " + idade + " anos.";
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Digite seu nome: ");
        String nome = sc.nextLine();

        System.out.print("Digite sua idade: ");
        int idade = sc.nextInt();

        // Versão 1 - Procedural
        System.out.println("\n--- Versão Procedural ---");
        System.out.println("Seja bem-vindo(a), " + nome + "! Você tem " + idade + " anos.");

        // Versão 2 - Modular
        System.out.println("\n--- Versão com Função ---");
        String mensagem = gerarMensagemBoasVindas(nome, idade);
        System.out.println(mensagem);

        sc.close();
    }
}

File: Funções/FunçõesPython/BoasVindas.py
# Versão 1 - Procedural

nome = input("Digite seu nome: ")
idade = int(input("Digite sua idade: "))

print(f"Seja bem-vindo(a), {nome}! Você tem {idade} anos.")


# Versão 2 - Modular com Funções

def gerar_mensagem_boas_vindas(nome, idade):
    return f"Seja bem vindo! {nome} sua idade é {idade}"

# Programa principal

nome = input("Digite seu nome: ")
idade = int(input("Digite sua idade: "))
mensagem = gerar_mensagem_boas_vindas(nome, idade)
print(mensagem)
